### PR TITLE
Add Azure Devops connector

### PR DESCRIPTION
**Pull request by joostmeijles**
This implements #32 

Example command:
```
PS> .\Beacon.exe azuredevops --url https://dev.azure.com/avivasolutions-public --project Nyxie --definitionid 10 -r
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,14 +16,29 @@ The easiest deployment mechanism is to install [Chocolatey](https://chocolatey.o
...
 
 ## Example usage
+### Azure Devops
+Using a public Azure Devops project and running continuously:
+
+    beacon azuredevops --url https://dev.azure.com/my-public-project --project MyProject --definitionid 1
+
+Or, using a personal access token (e.g. 1234567890ab, see [here](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page) for how to obtain a token) "for a private project and a custom branch name "branch/my":
 
+    beacon azuredevops --url https://dev.azure.com/my-public-project --project MyProject --definitionid 1 --token 1234567890ab -b -b refs/heads/branch/my
```
Comment (dennisdoomen) [inline README.md:26]: Does it support multiple definition IDs?

File: Sources/Beacon.Core/AzureDevopsConfig.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace Beacon.Core
+{
+    public class AzureDevopsConfig : Config
```
Comment (dennisdoomen) [inline Sources/Beacon.Core/AzureDevopsConfig.cs:5]: (nitpick) I think you should spell it as `AzureDevOps`.

File: README.md (hunk 1 above)
Comment (dennisdoomen) [reply README.md:26]: I see `-b` twice. Is that a typo?

File: Sources/Beacon.Core/Beacon.Core.csproj [hunk 3]
```diff
@@ -25,7 +25,7 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <Prefer32Bit>false</Prefer32Bit>
-    <LangVersion>7</LangVersion>
+    <LangVersion>6</LangVersion>
```
Comment (dennisdoomen) [inline Sources/Beacon.Core/Beacon.Core.csproj:28]: Why did you downgrade the language version?

File: Sources/Beacon.Core/Beacon.Core.csproj [hunk 4]
```diff
@@ -54,9 +54,117 @@
...
+    </Reference>
+    <Reference Include="System.Net.Http.Primitives, Version=4.2.29.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
+      <HintPath>..\packages\System.Net.Http.Formatting.Extension.5.2.3.0\lib\System.Net.Http.Primitives.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="System.Net.Http.WebRequest, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
+      <HintPath>..\packages\System.Net.Http.2.0.20126.16343\lib\net40\System.Net.Http.WebRequest.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="System.Runtime.Serialization" />
```
Comment (dennisdoomen) [inline Sources/Beacon.Core/Beacon.Core.csproj:167]: Do we really need _all_ these dependencies?

File: Sources/Beacon/Program.cs [hunk 5]
```diff
@@ -58,13 +61,31 @@ private static void RunTeamCityMonitor(Options options)
...
+        }
+        
+        private static void HandleParseErrors(ParserResult<object> result, IEnumerable<Error> errors)
         {
             var helpText = new HelpText
             {
                 AddDashesToOption = true,
                 AdditionalNewLineAfterOption = true,
-                Copyright = new CopyrightInfo("Dennis Doomen", 2015, 2018).ToString(),
+                Copyright = new CopyrightInfo("Dennis Doomen", 2015, 2020).ToString(),
```
Comment (dennisdoomen) [inline Sources/Beacon/Program.cs:88]: I guess we should change it to `Aviva Solutions`.

**Comment by dennisdoomen** [comment]
Oh, and did you verify the final ZIP/chocolatey package has all the dependencies?

File: README.md (hunk 1 above)
Comment (joostmeijles) [reply README.md:26]: > Does it support multiple definition IDs?

No

File: Sources/Beacon.Core/Beacon.Core.csproj (hunk 3 above)
Comment (joostmeijles) [reply Sources/Beacon.Core/Beacon.Core.csproj:28]: The PSake build did not work with 7. Did not investigate it a lot deeper, took the easy way and downgraded it.

**Comment by joostmeijles** [comment]
> Oh, and did you verify the final ZIP/chocolatey package has all the dependencies?

Verified that the azuredevops command works in the `Beacon\Package\Output` directory by running:
```
.\Beacon.exe azuredevops  --url https://dev.azure.com/avivasolutions-public --project Nyxie --definitionid 10
```

File: Sources/Beacon.Core/Beacon.Core.csproj (hunk 4 above)
Comment (joostmeijles) [reply Sources/Beacon.Core/Beacon.Core.csproj:167]: Almost, removed 1.

File: README.md (hunk 1 above)
Comment (dennisdoomen) [reply README.md]: At my client, we have usually have about 10 builds in different folders that we monitor.

File: Sources/Beacon.Core/Beacon.Core.csproj (hunk 3 above)
Comment (dennisdoomen) [reply Sources/Beacon.Core/Beacon.Core.csproj:28]: Ah, ok. I remember that problem. I'll fix that later.

File: README.md (hunk 1 above)
Comment (joostmeijles) [reply README.md]: I could add support for multiple pipelines (i.e. definition ids) per Azure DevOps project, e.g. https://dev.azure.com/avivasolutions-public/Nyxie/_build has 2 pipelines that could be monitored with the same build light.

What do you think?

Comment (dennisdoomen) [reply README.md]: That would be very useful IMO and consistent with the TeamCity implementation.

Comment (joostmeijles) [reply README.md]: Implemented: https://github.com/dennisdoomen/Beacon/pull/33/commits/9c23333d5fd7ee37687f6e9097a6b558b0966a27

**Pull request merged**